test/test_automl/test_automl.py

```diff
@@ -114,6 +115,33 @@ def test_fit(self):
         self._tearDown(backend_api.temporary_directory)
         self._tearDown(backend_api.output_directory)
 
+    def test_delete_non_winning_models(self):
+        backend_api = self._create_backend(
+            'test_delete', delete_tmp_folder_after_terminate=False)
+
+        X_train, Y_train, _, _ = putil.get_dataset('iris')
+        automl = autosklearn.automl.AutoML(
+            backend_api,
+            time_left_for_this_task=20,
+            per_run_time_limit=5,
+            ensemble_nbest=3,
+        )
+        automl.fit(
+            X_train, Y_train, metric=accuracy, task=MULTICLASS_CLASSIFICATION,
+        )
+
+        model_files = glob.glob(os.path.join(
+            backend_api.temporary_directory, '.auto-sklearn', 'models', '*.model'))
+
+        # Assert that the files of the models used by the ensemble are available
+        ensemble_members = set([idx[1] for idx in automl.ensemble_.identifiers_])
+        model_files = set([int(m_file.split('.')[-2]) for m_file in model_files])
+        self.assertSetEqual(ensemble_members & model_files, ensemble_members)
```

Comment:
Could you please evaluate whether it's possible to check that some files are deleted? Maybe with this wrap method? https://stackoverflow.com/a/25608441

Reply:
I tried to do this using a wrap, but was stumbling on some problems. The issue is that the EmsembleBuilder inside AutoML (automl.proc_ensemble) is purged at the end of the fit() function. I could get around this, but I would have to add some strange code just to make the test/wrapping work. Therefore, I opted by another more simple solution.

Since I'm logging the model deletions, I think the easiest way to check for deletions would be to inspect the log. That is what I implemented.

Let me know what your thoughts on this.